src/charm.py

```diff
@@ -35,83 +79,493 @@ def __init__(self, *args: typing.Any):
+                summary="OpenCTI platform/worker",
+                description="OpenCTI platform/worker",
+                services={
+                    "charm-callback": {
+                        "override": "replace",
+                        "command": f"bash {self._install_callback_script(health_check_url)}",
+                    },
+                    "platform": {
+                        "override": "replace",
+                        "command": "node build/back.js",
+                        "working-dir": "/opt/opencti",
+                        "environment": {
+                            "NODE_OPTIONS": "--max-old-space-size=8096",
+                            "NODE_ENV": "production",
+                            "PYTHONUNBUFFERED": "1",
+                            "APP__PORT": "8080",
+                            "APP__APP_LOGS__LOGS_LEVEL": "info",
+                            "PROVIDERS__LOCAL__STRATEGY": "LocalStrategy",
+                            "APP__TELEMETRY__METRICS__ENABLED": "true",
+                            **self._gen_secret_env(),
+                            **self._prepare_opensearch_env(),
+                            **self._gen_rabbitmq_env(),
+                            **self._gen_redis_env(),
+                            **self._gen_s3_env(),
+                            **self._gen_ingress_env(),
+                        },
+                    },
+                    "worker-0": worker_service,
+                    "worker-1": worker_service,
+                    "worker-2": worker_service,
+                },
+            ),
+            combine=True,
+        )
+        self._container.replan()
+        self._container.start("platform")
+        try:
+            self._check_platform_health(health_check_url)
+        except PlatformNotReady as exc:
+            self._container.start("charm-callback")
+            raise PlatformNotReady("waiting for opencti platform to start") from exc
+        self._container.stop("charm-callback")
+        self._container.add_layer(
+            label="opencti",
+            layer=ops.pebble.LayerDict(
+                summary="OpenCTI platform/worker",
+                description="OpenCTI platform/worker",
+                checks={
+                    "platform": {
+                        "override": "replace",
+                        "level": "ready",
+                        "http": {"url": health_check_url},
+                        "period": "1m",
+                        "timeout": "5s",
+                        "threshold": 5,
+                    }
+                },
+            ),
+            combine=True,
+        )
+        self._container.replan()
+        self._container.start("worker-0")
+        self._container.start("worker-1")
+        self._container.start("worker-2")
 
-        Learn more about config at https://juju.is/docs/sdk/config
+    def _install_callback_script(self, health_check_url: str) -> pathlib.Path:
```

Comment:
I think it might be more efficient if we have a review call for this, could you organise something with me and @gregory-schiano please

Reply:
Yes, no problem, created.